File: for-local-test/python/228.summary-ranges.py
class Solution:
    def summaryRanges(self, nums: list[int]) -> list[str]:
        answer = []
        if len(nums) == 0:
            return answer
        start = nums[0]
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1] + 1:
                if start == nums[i - 1]:
                    answer.append(str(start))
                else:
                    answer.append(str(start) + "->" + str(nums[i - 1]))
                start = nums[i]
        if start == nums[-1]:
            answer.append(str(start))
        else:
            answer.append(str(start) + "->" + str(nums[-1]))
        return answer
    
if __name__ == "__main__":
    solution = Solution()
    nums = [0,1,2,4,5,7]
    print(solution.summaryRanges(nums))

File: for-local-test/228.summary-ranges.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> summaryRanges(int[] nums) {
        if (nums == null || nums.length == 0) return new ArrayList<>();
        List<String> answer = new ArrayList<>();
        int n = nums.length;
        int start = nums[0];
        for (int i = 0; i < n; i++){
            if(i == n - 1 || nums[i + 1] != nums[i] + 1){
                if (start == nums[i]){
                    answer.add(String.valueOf(start));
                } else {
                    answer.add(start + "->" + nums[i]);
                }
                if (i != n - 1){
                    start = nums[i + 1];
                }
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums = {0,1,2,4,5,7};
        System.out.println(solution.summaryRanges(nums));
    }
}

